src/openrct2/drawing/Drawing.Sprite.cpp

```diff
@@ -282,6 +282,20 @@ bool gfx_load_g2()
         // Read element data
         _g2.data = fs.ReadArray<uint8_t>(_g2.header.total_size);
 
+        if (_g2.header.num_entries != G2_SPRITE_COUNT)
+        {
+            std::string errorMessage = "Mismatched g2.dat size.\nExpected: " + std::to_string(G2_SPRITE_COUNT) + "\nActual: "
```

Comment:
The message could maybe mention the user is about to experience graphical glitches due to this

Reply:
Resolve by adding prompt that appears after first prompt detailing issue and how to fix.